programs/gummyroll_crud/src/lib.rs

```diff
@@ -4,51 +4,149 @@ use gummyroll::{program::Gummyroll, Node};
 declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
 
 #[derive(Accounts)]
-pub struct Add<'info> {
+#[instruction(max_depth: u32, max_buffer_size: u32)]
+pub struct CreateTree<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(
+        init,
+        seeds = [
+            b"gummyroll-crud-authority-pda",
+            merkle_roll.key().as_ref(),
+            authority.key().as_ref(),
+        ],
+        bump,
+        payer = authority,
+        space = 0
+    )]
+    /// CHECK: This account is neither written to nor read from.
+    pub authority_pda: UncheckedAccount<'info>,
+    pub gummyroll_program: Program<'info, Gummyroll>,
     #[account(mut)]
     /// CHECK: unsafe
     pub merkle_roll: UncheckedAccount<'info>,
-    pub owner: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct Add<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(
+        seeds = [
+            b"gummyroll-crud-authority-pda",
+            merkle_roll.key().as_ref(),
+            authority.key().as_ref(),
+        ],
+        bump,
+    )]
+    /// CHECK: This account is neither written to nor read from.
+    pub authority_pda: UncheckedAccount<'info>,
     pub gummyroll_program: Program<'info, Gummyroll>,
+    #[account(mut)]
+    /// CHECK: unsafe
+    pub merkle_roll: UncheckedAccount<'info>,
 }
 
 #[derive(Accounts)]
 pub struct Remove<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
```

Comment:
I remember that you suggested the asset owner should be able to delete leaves, @jarry-xiao, but I made it so that only the tree owner can do so with the plain vanilla add/remove instructions. I thought it would be useful to just have some super-user power to add and remove things from the tree, if it's your tree.

When we have a good use case for an owner having that power (eg. a decompress operation) then maybe we can just add a new instruction?

Reply:
I think this totally depends on context, but we can probably use this same instruction and just check that the authority is either the owner or the super authority